fuzz/src/actions.rs

```diff
@@ -204,6 +206,19 @@ impl Action<'_> {
                 assert_eq!(control.as_str(), compact.as_str());
                 assert_eq!(compact.capacity(), compact_capacity);
             }
+            Remove(val) => {
+                let idx = to_index(control, val);
```

Comment:
`to_index()` can also return `s.len()`. This should have caused some crashes in the fuzzer.

Reply:
Good call out! It looks like as it exists today `to_index()` calls `.cycle()` before it `.chain([s.len()])`, which was causing `s.len()` to never get emitted, since `.cycle()` never ends. I updated `to_index()` to fix this and added tests to make sure it's doing what we want